components/picture_selector.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useEffect} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {localizeMessage} from 'utils/utils';
+import * as FileUtils from 'utils/file_utils';
+
+import './picture_selector.scss';
+
+type Props = {
+    src?: string;
+    defaultSrc?: string;
+    loadingPicture?: boolean;
+    onSelect: (file: File) => void;
+    onRemove: () => void;
+};
+
+const PictureSelector: React.FC<Props> = (props: Props) => {
+    const {src, defaultSrc, loadingPicture, onSelect, onRemove} = props;
+
+    const [image, setImage] = useState<string>();
+    const [orientationStyles, setOrientationStyles] = useState<{transform: any; transformOrigin: any}>();
+
+    const inputRef: React.RefObject<HTMLInputElement> = React.createRef();
+
+    const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (e.target.files && e.target.files[0]) {
+            const file = e.target.files[0];
+
+            const reader = new FileReader();
+            reader.onload = (ev) => {
+                setImage(URL.createObjectURL(file));
+
+                const orientation = FileUtils.getExifOrientation(ev.target!.result);
+                setOrientationStyles(FileUtils.getOrientationStyles(orientation));
+            };
+            reader.readAsArrayBuffer(file);
+
+            onSelect(file);
+        }
+    };
+
+    const handleInputFile = () => {
+        if (!inputRef || !inputRef.current) {
+            return;
+        }
+
+        inputRef.current.value = '';
+        inputRef.current.click();
+    };
+
+    const handleRemove = () => {
+        onRemove();
+        if (defaultSrc) {
+            setImage(defaultSrc);
+        } else {
+            setImage(undefined);
+        }
+    };
+
+    useEffect(() => {
+        if (!image) {
+            if (src) {
+                setImage(src);
+            } else if (defaultSrc) {
+                setImage(defaultSrc);
+            }
+        }
+    }, [src, setImage]);
```

Comment:
React complains if you don’t provide dependencies for everything external. I wonder why image  isn’t in dependencies?

Reply:
@reflog had me add `setImage`, shouldn't that suffice?

Reply:
- `setImage` isn't the true state change but a method, useEffect does a comparison with the value. 
- you can definitely add it, but it would raise complexity and create new questions, because it raises the issue of comparing and wondering of the value of the function returned from the state.

Reply:
I'm confused, is there anything I should change here? Still new to hooks lol.

Reply:
```suggestion
    }, [src, image]);
```